### DISCUSSION TITLE
How to debug "System.ExeuctionEngineException"

### DISCUSSION THREAD

**Discussion by captainkidd5** [amerkoleci/JoltPhysicsSharp#106] [Announcements]
I'm currently working on a commercial game and thus can't share the project for reproduction, but I'll need to be able to iron out these seemingly random System.ExeuctionEngineException errors somehow. 

Would it be possible to somehow expose the underlying error to a debugger?

`System.ExecutionEngineException: 'Exception of type 'System.ExecutionEngineException' was thrown.'`

in the console:

`Fatal Error. Internal CLR Error. (0x80131506)`

**Comment by amerkoleci** [comment]
Please avoid opening issues that are not able to be reproduced thus fixed. If you have a repro case I'll gladly give a look.

**Reply by captainkidd5** [reply]
Apologizes. I'll see if I can make one.

**Comment by captainkidd5** [comment]
@amerkoleci I've tried to reproduce the other issue I was having, which I have determined has come from raycasting, although I can't get the same issue to occur in this repo.

In the meantime, here is a working small sample in Monogame. I have found another issue when attempting to detect whether or not a body has been added to the simulation (or even created).

[see here](https://github.com/captainkidd5/Jolt-Physics-Monogame-Test/blob/6abca69fb088167adfd265bc5a1b8b975b369eae/Tests.cs#L48)


There is also an issue [here](https://github.com/captainkidd5/Jolt-Physics-Monogame-Test/blob/6abca69fb088167adfd265bc5a1b8b975b369eae/Tests.cs#L128) which crops up in maybe 1 in 50 debugger runs in visual studio, although I can't get it to reliably occur, and it may have something to do with pausing the debugger. I'll keep looking out for that one, but in the meantime let me know if it crashes for you.

**Reply by amerkoleci** [reply]
Hi,
Thanks for bringing an example, Body is class now and handles logic as NativeObject, fixed the memory issue as well.

Please grab the `2.11.0` package.

**Reply by captainkidd5** [reply]
Amazing!! Thank you!! This appears to have solved the other issues I was having as well!

**Comment by amerkoleci** [comment]
Nice, there was an issue between native and managed memory, fortunately solved now
